owners/src/ownership/parser.js

```diff
@@ -261,6 +261,23 @@ class OwnersParser {
       }
     }
 
+    // Check that there are no singleton patterns in brace-expansion format. The
+    // minimatch glob parsing library performs brace-expansion, but only if the
+    // braces contain comma-separated patterns. This is not obvious behavior,
+    // and has resulted in broken owners rules.
+    // Ex. `{a,b}.js` becomes /(a|b)\.js/
+    // Ex. `{a}.js` becomes /\{a\}\.js/
+    if (/\{[^,]+\}/.test(pattern)) {
```

Comment:
New edge-case introduced in this PR: what if the file actually includes `{` and `}` in its name? 😅

Reply:
Then it is wrong :laughing:

Reply:
But actually that should be disallowed and I'm comfortable with not supporting that at present lmao